src/js/components/modals/UpdateConfigModalContents.js

```diff
@@ -0,0 +1,177 @@
+      this.props.setPendingRequest(true);
+    }
+  }
+
+  getConfigSubmit(triggerSubmit) {
+    this.triggetConfigSubmit = triggerSubmit;
+  }
+
+  getUpdateErrorContent() {
+    let {updateError} = this.props;
+
+    return (
+      <CosmosErrorMessage
+        className="text-error-state text-overflow-break-word text-align-center"
+        error={updateError}
+        wrapperClass="modal-content-inner container container-pod
+          container-pod-short horizontal-center" />
+    );
+  }
+
+  getMarathonConfigurationModalContents() {
+    // TODO: The marathon configuration modal content needs to be returned.
+    // https://mesosphere.atlassian.net/browse/DCOS-6639
+    return null;
+  }
+
+  getPackageConfigurationModalFooter() {
+    let {hasFormErrors} = this.state;
+    let {pendingRequest, updateError, updateSuccess} = this.props;
+
+    if (updateError) {
+      return (
+        <div className="button-collection horizontal-center flush-bottom">
+          <button
+            disabled={pendingRequest}
+            className="button flush-bottom button-wide"
+            onClick={this.props.handleErrorConfigEditClick}>
+            Edit Configuration
+          </button>
+        </div>
+      );
+    }
+
+    if (updateSuccess) {
+      return (
+        <div className="button-collection horizontal-center flush-bottom">
+          <button
+            disabled={pendingRequest}
+            className="button button-success flush-bottom button-wide"
+            onClick={this.props.onClose}>
+            <i className="icon icon-sprite icon-sprite-mini
+              icon-sprite-mini-white icon-check-mark" />
+          </button>
+        </div>
+      );
+    }
+
+    return (
+      <div className="button-collection flush-bottom">
+        <button className="button button-large"
+          onClick={this.props.onClose}>
+          Back
+        </button>
+        <button disabled={hasFormErrors}
+          className="button button-large button-success"
+          onClick={this.handleConfigSave}>
+          Save Configuration
+        </button>
+      </div>
+    );
+  }
+
+  getPackageConfigurationModalContents(service) {
+    // TODO: The currentConfiguration needs to be populated with the package's
+    // current configuration.
+    // https://mesosphere.atlassian.net/browse/DCOS-6638
+    let currentConfiguration = {};
+    let {name, version} = service.getPackage();
+    let icon = service.getIcons()['icon-small'];
+    let schema = service.getConfig();
+
+    return (
+      <AdvancedConfig
+        model={currentConfiguration}
+        packageIcon={icon}
+        packageName={name}
+        packageVersion={version}
+        schema={schema}
+        onChange={this.handleAdvancedFormChange}
+        getTriggerSubmit={this.getConfigSubmit} />
+    );
+  }
+
+  render() {
+    let {servicePackage, updateError} = this.props;
+    let modalContent;
+
+    if (!servicePackage) {
+      return null;
```

Comment:
Not here, because it's possible that `servicePackage` doesn't exist, and we don't want to return an error because then the modal will render with that content always.